### PR TITLE
fix: check the disk capacity

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind bug
/kind failing-test

**What this PR does / why we need it**:

- set the default disk capacity
- compare the disk capacity when creating the disk

**Which issue(s) this PR fixes**:

Fixes #46 

**Special notes for your reviewer**:

**Release note**:
```
fix: compare the disk capacity
```

### CODE REVIEW

File: pkg/azuredisk/azuredisk.go [hunk 1]
```diff
@@ -358,3 +358,11 @@ func checkDiskName(diskName string) bool {
 
 	return true
 }
+
+func getResourceGroupFromDiskURI(diskURI string) (string, error) {
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go]: pls also copy the unit test from cloud provider

File: pkg/azuredisk/controllerserver.go [hunk 2]
```diff
@@ -68,9 +69,12 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	if !isValidVolumeCapabilities(volumeCapabilities) {
 		return nil, status.Error(codes.InvalidArgument, "Volume capabilities not supported")
 	}
-
-	volSizeBytes := int64(req.GetCapacityRange().GetRequiredBytes())
+	capacityBytes := req.GetCapacityRange().GetRequiredBytes()
+	volSizeBytes := int64(capacityBytes)
 	requestGiB := int(volumehelper.RoundUpGiB(volSizeBytes))
+	if requestGiB == 0 {
+		requestGiB = 32
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go]: define a const: `defaultDiskSize = 1`

File: pkg/azuredisk/controllerserver.go [hunk 3]
```diff
@@ -205,6 +209,20 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		}
 	}
 
+	diskName = path.Base(diskURI)
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go]: is it ok to add a standalone func to check existence and capacity of azure disk?

File: pkg/azuredisk/azuredisk.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: I have removed the function.

File: pkg/azuredisk/controllerserver.go (hunk 2 above)
Comment (ZeroMagic) [reply pkg/azuredisk/controllerserver.go]: done.

File: pkg/azuredisk/azuredisk.go [hunk 4]
```diff
@@ -49,6 +52,8 @@ const (
 	// see https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
 	diskNameMinLength = 1
 	diskNameMaxLength = 80
+
+	defaultDiskSize = 1
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go]: add comment
```
defaultDiskSize = 1  #GiB
```

File: pkg/azuredisk/azuredisk.go [hunk 5]
```diff
@@ -174,6 +179,16 @@ func (d *Driver) checkDiskExists(ctx context.Context, diskURI string) error {
 	return nil
 }
 
+func (d *Driver) checkDiskCapacity(ctx context.Context, resourceGroup, diskName string, requestGiB int) (bool, error) {
+	disk, err := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go]: add condition:
```
if err != nil { ...}
```

File: pkg/azuredisk/azuredisk.go [hunk 6]
```diff
@@ -174,6 +179,16 @@ func (d *Driver) checkDiskExists(ctx context.Context, diskURI string) error {
...
+	if err == nil {
+		if disk.DiskSizeGB != nil && int(*disk.DiskSizeGB) != requestGiB {
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go]:
```
if disk != nil && disk.DiskSizeGB != nil && int(*disk.DiskSizeGB) != requestGiB {
```

Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: done.

File: pkg/azuredisk/azuredisk.go (hunk 5 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: done.

File: pkg/azuredisk/azuredisk.go (hunk 4 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: done.

**Pull request merged**